sdk/azidentity/chained_token_credential.go

```diff
@@ -62,36 +64,40 @@ func (c *ChainedTokenCredential) GetToken(ctx context.Context, opts policy.Token
 		token, err = cred.GetToken(ctx, opts)
 		var credErr credentialUnavailableError
 		if errors.As(err, &credErr) {
-			errList = append(errList, credErr)
+			errList = append(errList, err.(credentialUnavailableError))
```

Comment:
We already have a variable of the correct type
```suggestion
			errList = append(errList, credErr)
```

Reply:
wait but credErr is empty! no? 👀

Reply:
Oooh! https://go.dev/blog/go1.13-errors